src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala

```diff
@@ -482,6 +489,15 @@ class VerilogEmitter extends SeqTransform with Emitter {
     def getConnectEmissionOption(target: ReferenceTarget): ConnectEmissionOption =
       connectEmissionOption(target)
 
+    // Defines the memory initialization based on the annotation
+    // Defaults to having the memories inside the `ifndef SYNTHESIS` block
+    def emitMemoryInitAsNoSynth: Boolean = {
+      annotations.collectFirst {
+        case MemoryNoSynthInit => true
+        case MemorySynthInit   => false
+      }.getOrElse(true)
+    }
```

Comment:
```suggestion
      val annos = annotations.collect { case a @ (MemoryNoSynthInit | MemorySynthInit) => a }
      annos match {
        case Seq() => true
        case Seq(MemoryNoSynthInit) => true
        case Seq(MemorySynthInit) => false
        case other =>
          throw new FirrtlUserException("There should only be at most one memory initialization option annotation, got $other")
      }
    }
```
I'm not sure of the best way to express this, but we need to check that the user doesn't provide contradictory options, here I'm checking only 1 is specified but we could also `.distinct` the `annos` to allow more than 1 of the same kind.

Reply:
Yes, that's much better. I think for now there is no need for multiple values. Would over-complicate things.